net/netstat/netstat_windows.go

```diff
@@ -249,6 +250,12 @@ func moduleInfo[entryType moduleInfoConstraint](entry *entryType, proc *windows.
 	}
 
 	basicInfo := (*_TCPIP_OWNER_MODULE_BASIC_INFO)(addr)
+	// GetOwnerModuleFromTcp*Entry is apparently using nil as an empty result
+	// under certain circumstances, so we check all the things.
+	if basicInfo == nil || basicInfo.moduleName == nil {
```

Comment:
oh, I just noticed this latter check is unnecessary, as UTF16PtrToString is documented to handle nil:

```go
// UTF16PtrToString takes a pointer to a UTF-16 sequence and returns the corresponding UTF-8 encoded string.
// If the pointer is nil, it returns the empty string. It assumes that the UTF-16 sequence is terminated
// at a zero word; if the zero word is not present, the program may crash.
func UTF16PtrToString(p *uint16) string {
	if p == nil {
		return ""
	}
	if *p == 0 {
		return ""
	}

	// Find NUL terminator.
	n := 0
	for ptr := unsafe.Pointer(p); *(*uint16)(ptr) != 0; n++ {
		ptr = unsafe.Pointer(uintptr(ptr) + unsafe.Sizeof(*p))
	}

	return string(utf16.Decode(unsafe.Slice(p, n)))
}
```